checker_linux.go

```diff
@@ -145,8 +145,19 @@ func (c *Checker) CheckAddrZeroLinger(addr string, timeout time.Duration, zeroLi
 	if err != nil {
 		return err
 	}
+
+	var domain int
+	switch rAddr.(type) {
+	case *unix.SockaddrInet4:
+		domain = unix.AF_INET
+	case *unix.SockaddrInet6:
+		domain = unix.AF_INET6
+	default:
+		return errors.New("Unrecognized IP address type")
+	}
+
```

Comment:
Alternatively, `parseSockAddr` could return `(addr unix.Sockaddr, domain int, err error)` and the type switch can be avoided. I'm neutral on one way or the other and opted to start by *not* altering the signature.